internal/cluster/cluster_operations.go

```diff
@@ -462,20 +461,61 @@ func getFlattenedHostSpecsForRefs(ctx context.Context, hostRefs []vcf.HostRefere
 			api_client.LogError(vcfErr)
 			return nil, errors.New(*vcfErr.Message)
 		}
-		flattenedHostSpecs = append(flattenedHostSpecs, *FlattenHost(*hostObj))
+		flattenedHostSpecs = append(flattenedHostSpecs, FlattenHost(*hostObj))
 	}
 	return flattenedHostSpecs, nil
 }
 
-func getFlattenedVdsSpecsForRefs(vdsSpecs []vcf.VdsSpec) []map[string]interface{} {
-	flattenedVdsSpecs := *new([]map[string]interface{})
-	// Since backend API returns objects in random order sort VDSSpec list to ensure
-	// import is reproducible
-	sort.SliceStable(vdsSpecs, func(i, j int) bool {
-		return vdsSpecs[i].Name < vdsSpecs[j].Name
-	})
-	for _, vdsSpec := range vdsSpecs {
-		flattenedVdsSpecs = append(flattenedVdsSpecs, network.FlattenVdsSpec(vdsSpec))
+func getFlattenedVdsSpecsForRefs(ctx context.Context, clusterId string, apiClient *vcf.ClientWithResponses) ([]map[string]interface{}, error) {
+	vdsResponse, err := apiClient.GetVdsesWithResponse(ctx, clusterId)
+	if err != nil {
+		return nil, err
 	}
-	return flattenedVdsSpecs
+
+	if vdsResponse.JSON200 == nil {
+		return nil, fmt.Errorf("vdsResponse.JSON200 is nil")
+	}
+
+	flattenedVdsSpecs := make([]map[string]interface{}, len(*vdsResponse.JSON200))
+	for i, vds := range *vdsResponse.JSON200 {
+		portGroups := vds.PortGroups
+		portGroupSpecs := make([]vcf.PortgroupSpec, len(*portGroups))
+		for j, pg := range *portGroups {
+			portGroupSpecs[j] = vcf.PortgroupSpec{
+				Name:          pg.Name,
+				TransportType: pg.TransportType,
+				ActiveUplinks: pg.ActiveUplinks,
+			}
+		}
```

Comment:
@spacegospod - I recall we made some changes to avoid using JSON200 when we bumped to v0.4.1 of the SDK. Can you suggest an update for this section?

Reply:
Sure thing, what you need is `GetResponseAs[YourType](RawResponse)`

For reference https://github.com/vmware/terraform-provider-vcf/blob/main/internal/api_client/sddc_manager_client.go#L137

```suggestion

   switches, vcfErr := api_client.GetResponseAs[whatever type vdsResponse.JSON200 is](vdsResponse)
	if vcfErr != nil {
		api_client.LogError(vcfErr)
		return errors.New(*vcfErr.Message)
	}
	
	...
	// continue processing `switches`
	...
```

What GetResponseAs does is it checks the response code and parses the raw response body (a byte array) into whatever generic type you provide (if the request is successful) or into vcf.Error (if the request has failed)

Reply:
@burnsjared0415 - the following should work:

```go
func getFlattenedVdsSpecsForRefs(ctx context.Context, clusterId string, apiClient *vcf.ClientWithResponses) ([]map[string]interface{}, error) {
	res, err := apiClient.GetVdsesWithResponse(ctx, clusterId)
	if err != nil {
		return nil, err
	}

	switcheObj, vcfErr := api_client.GetResponseAs[[]vcf.Vds](res)
	if vcfErr != nil {
		api_client.LogError(vcfErr)
		return nil, errors.New(*vcfErr.Message)
	}

	flattenedVdsSpecs := make([]map[string]interface{}, len(*switcheObj))
	for i, vds := range *switcheObj {
		portGroups := vds.PortGroups
		portGroupSpecs := make([]vcf.PortgroupSpec, len(*portGroups))
		for j, pg := range *portGroups {
			portGroupSpecs[j] = vcf.PortgroupSpec{
				Name:          pg.Name,
				TransportType: pg.TransportType,
				ActiveUplinks: pg.ActiveUplinks,
			}
		}

		niocBandwidthAllocations := vds.NiocBandwidthAllocations
		niocSpecs := make([]vcf.NiocBandwidthAllocationSpec, len(*niocBandwidthAllocations))
		for k, nioc := range *niocBandwidthAllocations {
			niocType := ""
			if nioc.Type != nil {
				niocType = *nioc.Type
			}

			niocSpecs[k] = vcf.NiocBandwidthAllocationSpec{
				Type: niocType,
				NiocTrafficResourceAllocation: vcf.NiocTrafficResourceAllocation{
					SharesInfo: &vcf.SharesInfo{
						Shares: nioc.NiocTrafficResourceAllocation.SharesInfo.Shares,
						Level:  nioc.NiocTrafficResourceAllocation.SharesInfo.Level,
					},
				},
			}
		}

		vdsSpec := vcf.VdsSpec{
			Name:                         vds.Name,
			IsUsedByNsxt:                 vds.IsUsedByNsxt,
			PortGroupSpecs:               &portGroupSpecs,
			NiocBandwidthAllocationSpecs: &niocSpecs,
		}

		flattenedVdsSpecs[i] = network.FlattenVdsSpec(&vdsSpec)
	}

	return flattenedVdsSpecs, nil
}
```

Reply:
Hey, Jared! Have you have a chance to [look at this comment](https://github.com/vmware/terraform-provider-vcf/pull/235#discussion_r1878563246)?

Reply:
No sir, i have not had time to look, we can cancel this work if you want and someone else can pick it up

Reply:
I'll pick it up from here and add commits to your fork if that's ok.

Reply:
Ammended in [d142e5e](https://github.com/vmware/terraform-provider-vcf/commit/d142e5ee5115f8c40f9a976b71f9c5aa7adcfbfd). Ready to test again.